pkg/util/testutil/conditions.go

```diff
@@ -0,0 +1,47 @@
+package testutil
+
+import (
+	"github.com/mongodb/mongodb-atlas-kubernetes/pkg/api/v1/status"
+	"github.com/onsi/gomega/format"
+	"github.com/onsi/gomega/types"
+)
+
+// MatchCondition returns the GomegaMatcher that checks if the 'actual' status.Condition matches the 'expected' one.
+func MatchCondition(expected status.Condition) types.GomegaMatcher {
+	return &conditionMatcher{ExpectedCondition: expected}
+}
+
+type conditionMatcher struct {
+	ExpectedCondition status.Condition
+}
+
+func (m *conditionMatcher) Match(actual interface{}) (success bool, err error) {
```

Comment:
this is the matcher that allows to use the power of matchers - it can be used to compare one condition against another or to find the condition in the array etc.